### PR TITLE
Add config option to stop chunkgen when above worldheight

### PR DESCRIPTION
**Pull request by ChimneySwift**
Config option to help reduce disk usage where players try to fly excessively far, often doing so above y=256

### CODE REVIEW

**Review by Proximyst — COMMENTED** [review]
What happens if you enable the option to collide on unloaded chunks?

**Comment by ghost** [comment]
@Proximyst The prevent unloaded chunks patch should be disabled above height limit, also, players above the height limit should not be checked for block collisions etc.

**Comment by MiniDigger** [comment]
should players even be allowed to escape the height limit?

**Comment by ChimneySwift** [comment]
> should players even be allowed to escape the height limit?

It's pretty easy to fly above the height limit in vanilla, and I've found many do so when they're trying to fly to particular coordinates. Perhaps that could be added in another config option but I don't personally see the need for it.

**Comment by ChimneySwift** [comment]
> What happens if you enable the option to collide on unloaded chunks?

Just had a chance to test new changes. Now if prevent entry into unloaded chunks is enabled along with this config option, it won't prevent entry if you're flying above the chunk, within the chunk it still behaves normally.

File: Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+ 
+     protected float getBlockJumpFactor() {
+         float f = this.world.getType(this.getChunkCoordinates()).getBlock().getJumpFactor();
+@@ -972,17 +984,24 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+                 for (int j = blockposition.getY(); j <= blockposition1.getY(); ++j) {
+                     for (int k = blockposition.getZ(); k <= blockposition1.getZ(); ++k) {
+                         blockposition_mutableblockposition.d(i, j, k);
+-                        IBlockData iblockdata = this.world.getType(blockposition_mutableblockposition);
+ 
+-                        try {
```
Comment (kashike) [inline Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch]: Please avoid changing the indentation of the existing code here.

Comment (ChimneySwift) [reply Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch]: I might be missing something here, but the indentation here is due to an added if statement. It looks like the comments are in the wrong spot however, which might be the actual issue here.

Comment (electronicboy) [reply Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch]: min-diff, changing the indentation causes more diff noise than is needed, causing un-needed code conflicts when merging in upstream changes, etc, makes the cost of keeping it pretty too high to be worth it

Comment (ChimneySwift) [reply Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch]: Ah I see what you mean.

Comment (ChimneySwift) [reply Spigot-Server-Patches/0604-Stop-entities-loading-chunks-when-running-checks.patch]: Believe latest commit should resolve this

**Comment by ChimneySwift** [comment]
Side effects/issues I've found so far:

- Elytra wear doesn't update while using above chunks
- Fireworks are unusually long lasting while flying above chunks
- Fireworks still generate chunks when flying above chunks

I'm not sure about the elytra issue, but I don't think the firework duration issue is directly caused by my changes, but rather because the firework entity isn't in a loaded chunk. EntityFireworks.explode() is run when it should but the entity isn't removed for a second or 2 after that. Potential workaround might just be to check if the entity is dead before ticking it, that's a bit ugly though and doesn't resolve generation issue.

**Pull request closed without merging**